lib/Support/ProgramCounterHandler.cpp

```diff
@@ -851,5 +851,13 @@ void PCH::buildHotPath(IRBuilder<> &B,
                                    CreateCmp(TypeCSV, Address.type()),
                                    CreateCmp(AddressCSV, Address.address()) };
   auto *Condition = B.CreateAnd(ToAnd);
-  B.CreateCondBr(Condition, BB, Default);
+
+  // Emit branches in distinct blocks.
+  auto *JumpToBB = BasicBlock::Create(B.getContext(), "", BB->getParent());
+  auto *JumpToDefault = BasicBlock::Create(B.getContext(), "", BB->getParent());
```

Comment:
1. Do you think we can create only one of the two?
2. Please elaborate in the comment why this is necessary. It won't be obvious in a month.

Reply:
1. I don't.
2. Elaborated.

Reply:
1. Why?
2. I don't see it. Don't mark comments as resolved unless you pushed the solution.

Reply:
My bad, wrong comment.

Reply:
1. Tested thoroughly, I keep triggering different asserts in IsolateFunctions, both removing either one or the other. I need the both targets to be in dedicated BBs to respect the boundaries used in IsolateFunctions.
2. Elaborated.

Reply:
I still don't see any additional comment...

Reply:
Should be fixed.

Reply:
The comment describes what the code does, not why it does it.
Why do we need to do this? Say that `FunctionIsolation` currently struggles with basic blocks that are both direct and indirect at the same time.

Reply:
Can you re-check now, please?